backend/routes/photos.js

```diff
@@ -21,21 +27,28 @@ router.put('/', async (req, res, next) => {
     try {
         if (req.body.photo_data) {
             const photoId = await req.models.photo.updatePhotoData(req.body.photoId, req.body.photo_data);
-            res.status(200).json(photoId);
-            next();
-        } else if (req.body.caption) {
+            const check = await req.models.photo.getPhoto(req.body.photoId);
+            res.status(200).json(check);
+        }
+        else if (req.body.caption) {
             const photoId = await req.models.photo.updatePhotoCaption(req.body.photoId, req.body.caption);
-            res.status(200).json(photoId);
-            next();
-        } else if (req.body.is_profile) {
+            const check = await req.models.photo.getPhoto(req.body.photoId);
+            res.status(200).json(check);
+        }
+        else if (req.body.is_profile) {
+
             if (req.body.is_profile != "1" && req.body.is_profile != "0") {
                 res.status(400).json({ message: "Invalid Input" });
```

Comment:
In a lot of places, I see we are throwing new Errors. On the frontend, those will show up as 500 Internal Server Errors which the frontend can't really do anything about. Basically, it will be an ungraceful ending. I think that using something like this, a status 400 with an error message. 4XX errors are client errors, ie bad syntax. 

You can read more here: https://en.wikipedia.org/wiki/List_of_HTTP_status_codes

If we return a 400 status instead of an actual Error, the behavior we have now will gracefully offer an alert box on the frontend, and allow the user to continue on the site/correct their information. You can customize the method to tell the user what went wrong too. But the throwing new Errors will bring down the whole website for the user.